pkgs/games/xash3d/wrapper.nix

```diff
@@ -0,0 +1,44 @@
+{ writeShellScriptBin
+, symlinkJoin
+, hlsdk
+, xash3d-unwrapped
+, name
+, gamedir ? name
+, basedir ? "$HOME/.local/share/xash3d"
+, dllname ? name # ARGH why couldn't this just be always name
+}:
+
+let
+  wrapper = writeShellScriptBin "xash3d-${name}" ''
+    export XASH3D_EXTRAS_PAK1=${xash3d-unwrapped}/share/xash3d/valve/extras.pk3
+
+    if [ -z "$XASH3D_BASEDIR" ]; then
+      export XASH3D_BASEDIR=${basedir}
+    fi
+
+    if [ ! -d "$XASH3D_BASEDIR" ]; then
+      echo "$XASH3D_BASEDIR not found"
+      exit 1
+    fi
```

Comment:
The potential error should be catched in nix

Reply:
Um no ? If you try to catch the error in Nix you're catching the error at build time. This is impossible because this here is a runtime error. If the directory `$XASH3D_BASEDIR` doesn't exist at runtime there's an error, but this is fine at build time. Especially that `$XASH3D_BASEDIR` can be changed by the user and so it can be anything at runtime.